tests/foreman/api/test_usergroup.py

```diff
@@ -0,0 +1,275 @@
+        """Update existing user group with different valid names.
+
+        @Feature: Usergroup
+
+        @Assert: User group is updated successfully.
+        """
+        user_group = entities.UserGroup().create()
+        for new_name in valid_data_list():
+            with self.subTest(new_name):
+                user_group.name = new_name
+                user_group = user_group.update(['name'])
+                self.assertEqual(new_name, user_group.name)
+
+    @tier1
+    def test_positive_update_with_new_user(self):
+        """Add new user to user group
+
+        @Feature: Usergroup
+
+        @Assert: User is added to user group successfully.
+        """
+        user = entities.User().create()
+        user_group = entities.UserGroup().create()
+        user_group.user.append(user)
+        user_group = user_group.update(['user'])
+        self.assertEqual(user.login, user_group.user[0].read().login)
+
+    @tier2
+    def test_positive_update_with_existing_user(self):
+        """Update user that assigned to user group with another one
+
+        @Feature: Usergroup
+
+        @Assert: User group is updated successfully.
+        """
+        users = [entities.User().create() for _ in range(2)]
+        user_group = entities.UserGroup(user=[users[0]]).create()
+        user_group.user[0] = users[1]
+        user_group = user_group.update(['user'])
+        self.assertEqual(users[1].login, user_group.user[0].read().login)
+
+    @tier1
+    def test_positive_update_with_new_role(self):
+        """Add new role to user group
+
+        @Feature: Usergroup
+
+        @Assert: Role is added to user group successfully.
+        """
+        new_role = entities.Role().create()
+        user_group = entities.UserGroup().create()
+        user_group.role.append(new_role)
+        user_group = user_group.update(['role'])
+        self.assertEqual(new_role.name, user_group.role[0].read().name)
+
+    @tier1
+    def test_positive_update_with_new_usergroup(self):
+        """Add new user group to existing one
+
+        @Feature: Usergroup
+
+        @Assert: User group is added to existing group successfully.
+        """
+        new_usergroup = entities.UserGroup().create()
+        user_group = entities.UserGroup().create()
+        user_group.usergroup.append(new_usergroup)
+        user_group = user_group.update(['usergroup'])
+        self.assertEqual(
+            new_usergroup.name, user_group.usergroup[0].read().name)
+
+    @tier1
+    def test_negative_update(self):
+        """Attempt to update existing user group using different invalid names.
+
+        @Feature: Usergroup
+
+        @Assert: User group is not updated.
+        """
+        user_group = entities.UserGroup().create()
+        for new_name in invalid_values_list():
+            with self.subTest(new_name):
+                user_group.name = new_name
+                with self.assertRaises(HTTPError):
+                    user_group = user_group.update(['name'])
```

Comment:
```python
                with self.assertRaises(HTTPError):
                    user_group.update(['name'])
```
`update()` will raise an exception, so no need in trying to set the value to some variable here.